G2-http3-protocol.md

```diff
@@ -0,0 +1,142 @@
+H3_ID_ERROR(0x0108)|n/a|INTERNAL
+H3_SETTINGS_ERROR(0x0109)|SETTINGS_TIMEOUT(4)|INTERNAL
+H3_MISSING_SETTINGS(0x010a)|SETTINGS_TIMEOUT(4)|INTERNAL
+H3_REQUEST_REJECTED(0x010b)|REFUSED_STREAM|UNAVAILABLE - Indicates that no processing occurred and the request can be retried, possibly elsewhere.
+H3_REQUEST_CANCELLED(0x010c)|CANCEL(8)|Mapped to call cancellation when sent by a client.Mapped to CANCELLED when sent by a server. Note that servers should only use this mechanism when they need to cancel a call but the payload byte sequence is incomplete.
+H3_REQUEST_INCOMPLETE(0x010d)|n/a|INTERNAL
+H3_MESSAGE_ERROR(0x010e)|n/a|INTERNAL
+H3_CONNECT_ERROR(0x010f)|CONNECT_ERROR|INTERNAL
+H3_VERSION_FALLBACK(0x0110)|n/a|INTERNAL
+n/a|FLOW_CONTROL_ERROR(3)|INTERNAL
+n/a|STREAM_CLOSED|No mapping as there is no open stream to propagate to. Implementations should log.
+n/a|COMPRESSION_ERROR|INTERNAL
+n/a|INADEQUATE_SECURITY| PERMISSION_DENIED … with additional detail indicating that permission was denied as protocol is not secure enough for call.
+
+#### Connection management
+
+`GOAWAY` and `PING` frames exist in HTTP/3 and serve the same purpose as in HTTP/2.
+One notable difference is the `GOAWAY` frame in HTTP/2 reports the last
+successfully processed stream ID. In HTTP/3 the `GOAWAY` frame ID value must be greater
+that the last successfully processed stream ID.
+
+### Exceeding deadlines
+
+When an RPC has exceeded its deadline in HTTP/2, the server will complete the response
+with a gRPC status code of DEADLINE_EXCEEDED, then send a `RST_STREAM` frame to the
+client to stop it from continuing to send the request stream. The client gets the
+response and reports the DEADLINE_EXCEEDED failure to the app.
+
+The `RST_STREAM` frame doesn't exist in HTTP/3. QUIC supports stopping the reading and sending parts of a QUIC
+stream independently. Two QUIC frames have replaced `RST_STREAM` functionality:
+
+* [RESET_STREAM](https://www.rfc-editor.org/rfc/rfc9000.html#name-reset_stream-frames) - Abruptly terminate the sending part of a stream.
+* [STOP_SENDING](https://www.rfc-editor.org/rfc/rfc9000.html#name-stop_sending-frames) - Requests that a peer cease transmission on a stream.
+
+`RESET_STREAM` is not appropriate because the response with the DEADLINE_EXCEEDED status
+code should reach the client. Because QUIC frames can be received out-of-order,
+sending this QUIC frame after completing the response can cause the client
+to abort the request.
+
+`STOP_SENDING` should be sent to the client by a gRPC server after it completes the response.
+It accurately communicates to the client that it should stop sending RPC messages.
+The application error code should be H3_NO_ERROR(0x0100).
+
+## Rationale
+
+HTTP/3 is a [highly requested feature](https://github.com/grpc/grpc/issues/19126) for gRPC.
+
+HTTP/3's benefits include:
```

Comment:
These bullets are just repeating the background. Remove them?

Rationale is really about "why did you make decision X instead of option Y." The highly requested issue and the mention of standardization is fine here, but normally I'd put it in background. (Background provides context and can explain why we are doing this feature.)